website/src/word-panel.tsx

```diff
@@ -1,195 +1,23 @@
-        <div>{p.segment.simplePhonetics}</div>
-      </>
-    )
-  }
-
-  return (
-    <div className={css.wordPanelContent}>
-      <IconButton
-        className={css.wordPanelButton.basic}
-        onClick={() => p.setContent(null)}
-        aria-label="Dismiss selected word information"
-      >
-        <MdClose size={32} />
-      </IconButton>
-      <header className={css.wordPanelHeader}>
-        <h1 className={css.noSpaceBelow}>Selected Word</h1>
-        <h2 className={css.cherHeader}>{p.segment.source}</h2>
-      </header>
-      <AudioPanel segment={p.segment} />
-      {phonetics ? (
-        <CollapsiblePanel
-          title={"Phonetics"}
-          content={phonetics}
-          icon={
-            <MdRecordVoiceOver
-              size={24}
-              className={css.wordPanelButton.colpleft}
-            />
-          }
-        />
-      ) : null}
-
-      {p.segment.segments?.length ? (
-        <CollapsiblePanel
-          title={"Word Parts"}
-          content={
-            <>
-              <VerticalMorphemicSegmentation
-                segments={p.segment.segments}
-                tagSet={p.tagSet}
-              />
-
-              {translation.length ? (
-                <div>&lsquo;{translation}&rsquo;</div>
-              ) : null}
-            </>
-          }
-          icon={
-            <IoEllipsisHorizontalCircle
-              size={24}
-              className={css.wordPanelButton.colpleft}
-            />
-          }
-        />
-      ) : null}
-      {p.segment.commentary ? (
-        <CollapsiblePanel
-          title={"Commentary"}
-          content={<>{p.segment.commentary}</>}
-          icon={<MdNotes size={24} className={css.wordPanelButton.colpleft} />}
-        />
-      ) : null}
-    </div>
-  )
-}
-
-export const VerticalMorphemicSegmentation = (p: {
-  segments: Dailp.FormFieldsFragment["segments"]
-  tagSet: TagSet
-}) => {
-  if (!p.segments) {
-    return null
-  }
-  if (p.segments) {
-    let segmentCount = p.segments.length
+  // Get information about the name of the first string in a list of features - this decides what type of information to grab from the FormFieldsFragment
+  const feature = props.features[0] as keyof FormFieldsFragment
 
-    return (
-      <table className={css.tableContainer}>
-        {p.segments.map((segment, index) => (
-          <tr>
-            <td className={css.tableCells}>
-              {index > 0 ? "-" : null}
-              {segment.morpheme}
-              {index !== segmentCount - 1 ? segment.nextSeparator : null}
-            </td>
-            <td className={css.tableCells}>
-              {segment.matchingTag ? segment.matchingTag.title : segment.gloss}
-            </td>
-          </tr>
-        ))}
-      </table>
-    )
+  // If the feature is an english gloss, it will return a special case of the component with quotes around the text
+  if (feature === "englishGloss") {
+    return <div>‘{props.word[feature]}’</div>
```

Comment:
This is another high cohesion problem, we should absolutely not be checking raw string equality. [This article is about Swift](https://academy.realm.io/posts/tryswift-andyy-hope-swift-eye-stringly-typed-api/) but I think a good enumeration of the problems with a "stringly typed API" like this one.

Reply:
Makes sense, I went ahead and changed this to add the quotation marks in the `PanelContent` component instead to reduce coupling.